### DISCUSSION TITLE
Login error: connection refused (xhost)

### DISCUSSION THREAD

**Discussion by vanillyn** [rustdesk/rustdesk#5095] [General]
### Bug Description

When trying to connect, I am prompted with the password (in this case, a permanent password that I had coordinated), and I am immediately logged out with the error "connection refused. Checking the logs, there are no errors of note.

### How to Reproduce

Simply attempt to connect using the ID.

### Expected Behavior

A valid connection to the server

### Operating system(s) on local side and remote side

Arch Linux -> Arch Linux

### RustDesk Version(s) on local side and remote side

1.2.1 -> 1.2.1

### Screenshots

![image](https://github.com/rustdesk/rustdesk/assets/108785622/2aa8baa8-68a8-449f-b0a3-d2634f45248a)

### Additional Context

I do know there have been countless other issues on this, all of which being resolved as fixed, but it seems it still doesn't work on Arch for me. If not, there may be another issue on my computer specifically.
Also interestingly, trying to connect to the same server on Windows 11 works, but not on Arch Linux.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/440

**Comment by xlionjuan** [comment]
I have encountered same issue on latest arch with KDE(x11) and 1.2.4 nightly

**Reply by xlionjuan** [reply]
But seems okay on Gnome and Openbox

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/issues/7324

**Reply by rustdesk** [reply]
Plasma6?

**Reply by xlionjuan** [reply]
Yes Plasma 6, Arch is a rolling distro, so every system's package version shouldn't has lot's of difference, except they didn't upgrade it

**Reply by rustdesk** [reply]
@xlionjuan could you try below 3 cases?

- [ ] login out / log in again
- [ ] reboot
- [ ] sudo service rustdesk restart

**Reply by xlionjuan** [reply]
Login, log out and reboot rarely fixed the issue, I said rarely, because it may ridiculously working sometimes

systemctl restart almost won't fix anything

Login screen don't have issues (sddm)

When using`pacman -U`(it means `dpkg -i`) upgrade and downgrade between 1.2.3 and 1.2.4 nightly, it may fix the issue for short period, maybe because it is doing systemctl start/stop service.


Additional information: if this happened while accepting connections, the window that can click accept or not accept will not show up, it may be an important clue.

**Comment by mac-linux-free** [comment]
but it works with the appimage version and plasma6.

**Reply by rustdesk** [reply]
AppImage is different story.

**Reply by xlionjuan** [reply]
for unattended access, still need to installed to system😅

**Comment by rustdesk** [comment]
Actually this `connection refused` is `x11 connection refused`. We did not fetch x11 var correctly from the system process.
AppImage and Flatpak do not have this issue, because they run in user mode.

**Comment by rustdesk** [comment]
@fufesou follow this up

**Comment by yangjuncode** [comment]
I also hit this in arch kde x11 session.

```
sudo pacman -S xorg-xhost
```
add below in my .bashrc solve my problem
```
xhost +local:root > /dev/null 2>&1
```

**Reply by rustdesk** [reply]
Thanks, this is a workaround of our bug not finding XAUTH, @xlionjuan you can try out this.

**Reply by xlionjuan** [reply]
@rustdesk and I found out the problem is gone without doing anything 😕

Edit: it happened again, and I install that package and added this line to `.bashrc` and reboot, it didn't happened, but I can't determine it isn't happening because of this.

**Reply by xlionjuan** [reply]
@rustdesk I tried another VM, I install it, and add it to `.bashrc`, and restart terminal and retry immediately, **without reboot**, it fixed.

**Reply by rustdesk** [reply]
Thanks

**Reply by Seqway** [reply]
> I also hit this in arch kde x11 session.
> 
> ```
> sudo pacman -S xorg-xhost
> ```
> 
> add below in my .bashrc solve my problem
> 
> ```
> xhost +local:root > /dev/null 2>&1
> ```

Is the problem not yet solved ? I have with actual version still this problem with Login failure and connection refused.
Only the entry in the -bashrc file solved it but that cant be the solution. Please check !

**Comment by rustdesk** [comment]
@xlionjuan we have a fix for this https://github.com/rustdesk/rustdesk/pull/7400, you can try out the nightly build with out `xhost` workaround.

**Reply by dgva** [reply]
EndeavourOS with KDE Plasma 6 (`nightly 1.2.4`), remoting from Windows 11 (`1.2.3`), it's working 100%!

**Comment by gusarg81** [comment]
Where I can download latest nightly which includes this fix? (In my case, .deb package).

Thanks.

EDIT: never mind.  I've found it. Sorry...

**Comment by Juniorsnet** [comment]
I have similar problem.
I use Manjaro with XFCE and light-locker, and rustdesk 1.3.1
Every time that the screen is locked (manually or by timeout) I lost connection.
I think that is related to light-locker
https://bugs.launchpad.net/ubuntu/+source/light-locker/+bug/1287171
If I start a vnc on display :1 and unlock screen. I can connect again with rustdesk.

**Comment by aptdnfapt** [comment]
i hosted this rust thing on a vps . and god how much time it took was insane . it was a breae bone fedora and dude bs doesnt work . 

but i did it . made this sh 
im not going to explain . cause this is bs . idc if u find this helpful or not . i have spent time and i dont want u too . or me ig . holy it was huff. 

#!/bin/bash 


sudo systemctl start lightdm.service
Xvfb :99 -screen 0 1024x768x24 &
export DISPLAY=:99
sudo systemctl restart rustdesk
startlxde &

**Comment by arpit2167407** [comment]
I am also using rustdesk in Linus to access the GPU, hoever, in between of the experiments it shows "Login Error, Connection Refused". Many times one of my colleague helped me by restarting the system but i woud like to know is there any permanent solution for this in order to do complete experiments smoothly.

**Comment by Seqway** [comment]
Is the problem not yet solved ? I have with actual version still this problem with Login failure and connection refused.
Only the entry in the -bashrc file solved it but that cant be the solution. Please check !

**Comment by markusantonuis** [comment]
I'm getting this issue on version 1.4.9 when trying to remote into my debian single board radxa computer. I don't have technical background suspect it may have something to do with the system not having an active display (despite displaying fine on the tv its' connected to) but there appears to be no display recognised in the system. Just wondering if this could have anything to do with why the connection is being refused.
